distributed/diagnostics/plugin.py

```diff
@@ -126,3 +128,54 @@ def transition(self, key, start, finish, **kwargs):
             Final state of the transition.
         kwargs: More options passed when transitioning
         """
+
+
+class PipInstall(WorkerPlugin):
+    """ A Worker Plugin to pip install a set of packages
+
+    This accepts a set of packages to install on all workers.
+    You can also optionally ask for the worker to restart itself after
+    performing this installation.
```

Comment:
I'd like to see a note of warning (here or elsewhere) that this is primarily intended for experimentation / interactive development. A more robust solution would be to include it in these in the actual worker environment (docker image, virtualenv/conda pack, etc.)